src/boot/linux.c

```diff
@@ -119,16 +93,72 @@ EFI_STATUS linux_exec(
         if (err != EFI_SUCCESS)
                 return log_error_status(err, "Bad kernel image: %m");
 
-        _cleanup_(unload_imagep) EFI_HANDLE kernel_image = NULL;
-        err = load_image(parent, kernel->iov_base, kernel->iov_len, &kernel_image);
+        if (secure_boot_enabled() && shim_loader_available())
```

Comment:
this should have a comment explaining the logic here. i.e. say that we go via BS in case of SHIM because shim wants to reauthenticate the payload, though i don't really get why, but whatever.